connectorx/src/sources/trino/mod.rs

```diff
@@ -0,0 +1,649 @@
+use std::{marker::PhantomData, sync::Arc};
+
+use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
+use fehler::{throw, throws};
+use prusto::{auth::Auth, Client, ClientBuilder, DataSet, Presto, Row};
+use serde_json::Value;
+use sqlparser::dialect::{Dialect, GenericDialect};
+use std::convert::TryFrom;
+use tokio::runtime::Runtime;
+
+use crate::{
+    data_order::DataOrder,
+    errors::ConnectorXError,
+    sources::Produce,
+    sql::{limit1_query, CXQuery},
+};
+
+pub use self::{errors::TrinoSourceError, typesystem::TrinoTypeSystem};
+use urlencoding::decode;
+
+use super::{PartitionParser, Source, SourcePartition};
+
+use anyhow::anyhow;
+
+pub mod errors;
+pub mod typesystem;
+
+#[throws(TrinoSourceError)]
+fn get_total_rows(rt: Arc<Runtime>, client: Arc<Client>, query: &CXQuery<String>) -> usize {
+    rt.block_on(client.get_all::<Row>(query.to_string()))
```

Comment:
It seems like this will execute the entire query and get all the results, which may be a bottleneck in terms of performance (since the query will be ran twice). Can we make it to using the `SELECT COUNT(*)` query to get the total rows? We have a util function named `count_query`. An example can be found [here](https://github.com/sfu-db/connector-x/blob/52da49ac0d2ddee458cc3279a20f5079b8b754f4/connectorx/src/sources/postgres/mod.rs#L70).